torch/_functorch/vmap.py

```diff
@@ -130,21 +131,33 @@ def _create_batched_inputs(
                       for in_dim, arg in zip(flat_in_dims, flat_args)]
     return tree_unflatten(batched_inputs, args_spec)
 
-# Undos the batching (and any batch dimensions) associated with the `vmap_level`.
 
+def _maybe_remove_batch_dim(name, batched_output, vmap_level, batch_size, out_dim):
+
+    if out_dim is None:
+        if isinstance(batched_output, torch.Tensor) and is_batchedtensor(batched_output):
+            raise ValueError(
+                f'vmap({name}, ...): `{name}` can not return a '
+                f'BatchedTensor when out_dim is None'
+            )
+        return batched_output
+
+    # out_dim is non None
+    if not isinstance(batched_output, torch.Tensor):
+        raise ValueError(f'vmap({name}, ...): `{name}` must only return '
+                         f'Tensors, got type {type(batched_output)} as a return. '
```

Comment:
I think `but got type {type(batched_output)}` should be enough i.e. no need for `as a return` (But not a strong opinion)

Reply:
removed